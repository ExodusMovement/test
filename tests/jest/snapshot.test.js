it('simple', () => {
  expect(10).toMatchSnapshot()
  expect(null).toMatchSnapshot()
  expect().toMatchSnapshot()
  expect([]).toMatchSnapshot()
  expect(/xx/).toMatchSnapshot()
  expect(Infinity).toMatchSnapshot()
  expect(false).toMatchSnapshot()
  expect(true).toMatchSnapshot()
  expect({}).toMatchSnapshot()
})

it('complex', () => {
  expect([10]).toMatchSnapshot()
  expect([{ a: 10 }]).toMatchSnapshot()
  expect({ a: 10 }).toMatchSnapshot()
  expect({ a: 10, b: 20 }).toMatchSnapshot()
  expect(Buffer.from('')).toMatchSnapshot()
})

// repeat test name
it('simple', () => {
  expect(/hello/).toMatchSnapshot()
  expect(true).toMatchSnapshot()
  expect(NaN).toMatchSnapshot()
  expect({}).toMatchSnapshot()
  expect(42).toMatchSnapshot()
  expect([]).toMatchSnapshot()
  expect(-Infinity).toMatchSnapshot()
})

it('mixed', () => {
  expect(true).toMatchSnapshot()
  expect([1, 2, 3]).toMatchSnapshot()
  expect({ foo: 'bar' }).toMatchSnapshot()
  expect(43).toMatchSnapshot()
  expect({}).toMatchSnapshot()
  expect([]).toMatchSnapshot()
})

// repeat test name
it('mixed', () => {
  expect([5, 4, 3]).toMatchSnapshot()
  expect([]).toMatchSnapshot()
  expect(false).toMatchSnapshot()
  expect(41).toMatchSnapshot()
  expect({ bar: 'buz' }).toMatchSnapshot()
  expect({}).toMatchSnapshot()
})

it('simple inline', () => {
  expect(42).toMatchInlineSnapshot(`42`)
  expect({}).toMatchInlineSnapshot(`{}`)
})

const TEST_ONE = { a: 20, d: Buffer.from('foo'), b: [1, 2, 'bar', 5], e: { foo: 'bar' } }
// eslint-disable-next-line no-sparse-arrays
const TEST_TWO = { ['__proto__']: [], b: [1, 2, , , 5], e: { foo: 'bar' }, f: -Infinity }
const TEST_THREE = [new Error('?!'), new TypeError('bar'), new Uint16Array([4, 2, 65_123]), null]

// Test names repeat on a purpose!

it('test A', () => {
  expect(TEST_ONE).toMatchSnapshot()
})

it('test B', () => {
  expect(TEST_TWO).toMatchSnapshot()
})

it('test B', () => {
  expect(TEST_THREE).toMatchSnapshot()
})

// Repeat name
it('test B', () => {
  expect({ x: 1337 }).toMatchSnapshot()
})

describe('nested test', () => {
  it('test A', () => {
    expect(TEST_TWO).toMatchSnapshot()
  })

  it('test A', () => {
    expect(TEST_THREE).toMatchSnapshot()
  })

  it('nested test one', () => {
    expect(TEST_ONE).toMatchSnapshot()
  })
})

it('test one, inline', () => {
  // eslint-disable-next-line unicorn/template-indent
  expect(TEST_ONE).toMatchInlineSnapshot(`
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`)
})

it('errors', () => {
  expect(() => {
    throw new TypeError('Whoops\nI\nFailed')
  }).toThrowErrorMatchingSnapshot()

  expect(() => {
    throw new RangeError('Out of something')
  }).toThrowErrorMatchingInlineSnapshot('"Out of something"')
})

it('async errors', async () => {
  await expect(Promise.reject(new TypeError('slow but nah'))).rejects.toThrowErrorMatchingSnapshot()
  await expect(Promise.resolve(new Error('ok'))).resolves.toThrowErrorMatchingInlineSnapshot('"ok"')
})

it('formateted objects', () => {
  expect(expect.any(Number)).toMatchSnapshot()
  expect({ a: expect.any(Number), b: expect.any(String) }).toMatchSnapshot()
})

describe('weird  names', () => {
  const ascii = Array.from({ length: 128 })
    .fill()
    .map((a, i) => i)
    .slice(0x20)
    .map((i) => String.fromCodePoint(i))
    .join('')
  for (const key of ['\n', '{}', '$', '`', '>', '` ` `', '` `\n `', '\\', '\\\n`\n\\``', ascii]) {
    it(key, () => {
      expect(key).toMatchSnapshot()
    })
  }

  it('multi\nline', () => {
    expect(0).toMatchSnapshot()
  })

  it('with `', () => {
    expect(42).toMatchSnapshot()
  })
})

// https://jestjs.io/docs/snapshot-testing#property-matchers

describe('property matchers', () => {
  it('will check the matchers and pass', () => {
    expect.assertions(1) // ensure we don't over-consume

    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'LeBron James',
    }

    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      id: expect.any(Number),
    })
  })

  it('will check the values and pass', () => {
    expect.assertions(1) // ensure we don't over-consume

    const user = {
      createdAt: new Date(),
      name: 'Bond... James Bond',
    }

    expect(user).toMatchSnapshot({
      createdAt: expect.any(Date),
      name: 'Bond... James Bond',
    })
  })
})

describe('', () => {
  it('', () => {
    expect('empty names test').toMatchSnapshot()
  })
})

test('deep matcher', () => {
  expect({
    one: 1,
    foo: {
      bar: {
        property: 'value 123',
        uuid: typeof crypto === 'undefined' ? 'hey Node.js 18' : crypto.randomUUID(),
      },
    },
  }).toMatchSnapshot({ foo: { bar: { uuid: expect.any(String) } } })
})

test('arrays', () => {
  expect([Math.random(), { a: 20 }]).toMatchSnapshot([expect.any(Number), {}])
  expect(['foo']).toMatchSnapshot([expect.any(String)])
  expect({ x: ['bar'], y: 10 }).toMatchSnapshot({ x: [expect.any(String)] })
  expect({ x: [{ a: 'baz', b: 20 }], y: 10 }).toMatchSnapshot({ x: [{ a: expect.any(String) }] })
  expect([{ x: [{ a: 'some', b: 20 }, { extra: 42 }], y: 10 }, { k: 1 }]).toMatchSnapshot([
    {
      x: [{ a: expect.any(String) }, {}],
    },
    {},
  ])
})

test('inline snapshots, prefixed', () => {
  // eslint-disable-next-line unicorn/template-indent
  expect({ a: 10, c: [1, 2], b: 20 }).toMatchInlineSnapshot(`
{
  "a": 10,
  "b": 20,
  "c": [
    1,
    2,
  ],
}
`)

  // eslint-disable-next-line unicorn/template-indent
  expect({ a: 10, c: [1, 2], b: 20 }).toMatchInlineSnapshot(`
  {
    "a": 10,
    "b": 20,
    "c": [
      1,
      2,
    ],
  }
          `) // end padding is ignored!
})
