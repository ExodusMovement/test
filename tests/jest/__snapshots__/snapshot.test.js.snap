exports[`  1`] = `
"empty names test"
`;

exports[`arrays 1`] = `
[
  Any<Number>,
  {
    "a": 20,
  },
]
`;

exports[`arrays 2`] = `
[
  Any<String>,
]
`;

exports[`arrays 3`] = `
{
  "x": [
    Any<String>,
  ],
  "y": 10,
}
`;

exports[`arrays 4`] = `
{
  "x": [
    {
      "a": Any<String>,
      "b": 20,
    },
  ],
  "y": 10,
}
`;

exports[`arrays 5`] = `
[
  {
    "x": [
      {
        "a": Any<String>,
        "b": 20,
      },
      {
        "extra": 42,
      },
    ],
    "y": 10,
  },
  {
    "k": 1,
  },
]
`;

exports[`async errors 1`] = `
"slow but nah"
`;

exports[`complex 1`] = `
[
  10,
]
`;

exports[`complex 2`] = `
[
  {
    "a": 10,
  },
]
`;

exports[`complex 3`] = `
{
  "a": 10,
}
`;

exports[`complex 4`] = `
{
  "a": 10,
  "b": 20,
}
`;

exports[`complex 5`] = `
{
  "data": [],
  "type": "Buffer",
}
`;

exports[`deep matcher 1`] = `
{
  "foo": {
    "bar": {
      "property": "value 123",
      "uuid": Any<String>,
    },
  },
  "one": 1,
}
`;

exports[`errors 1`] = `
"Whoops
I
Failed"
`;

exports[`formateted objects 1`] = `
Any<Number>
`;

exports[`formateted objects 2`] = `
{
  "a": Any<Number>,
  "b": Any<String>,
}
`;

exports[`mixed 1`] = `
true
`;

exports[`mixed 10`] = `
41
`;

exports[`mixed 11`] = `
{
  "bar": "buz",
}
`;

exports[`mixed 12`] = `
{}
`;

exports[`mixed 2`] = `
[
  1,
  2,
  3,
]
`;

exports[`mixed 3`] = `
{
  "foo": "bar",
}
`;

exports[`mixed 4`] = `
43
`;

exports[`mixed 5`] = `
{}
`;

exports[`mixed 6`] = `
[]
`;

exports[`mixed 7`] = `
[
  5,
  4,
  3,
]
`;

exports[`mixed 8`] = `
[]
`;

exports[`mixed 9`] = `
false
`;

exports[`nested test nested test one 1`] = `
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`;

exports[`nested test test A 1`] = `
{
  "__proto__": [],
  "b": [
    1,
    2,
    ,
    ,
    5,
  ],
  "e": {
    "foo": "bar",
  },
  "f": -Infinity,
}
`;

exports[`nested test test A 2`] = `
[
  [Error: ?!],
  [TypeError: bar],
  Uint16Array [
    4,
    2,
    65123,
  ],
  null,
]
`;

exports[`property matchers will check the matchers and pass 1`] = `
{
  "createdAt": Any<Date>,
  "id": Any<Number>,
  "name": "LeBron James",
}
`;

exports[`property matchers will check the values and pass 1`] = `
{
  "createdAt": Any<Date>,
  "name": "Bond... James Bond",
}
`;

exports[`simple 1`] = `
10
`;

exports[`simple 10`] = `
/hello/
`;

exports[`simple 11`] = `
true
`;

exports[`simple 12`] = `
NaN
`;

exports[`simple 13`] = `
{}
`;

exports[`simple 14`] = `
42
`;

exports[`simple 15`] = `
[]
`;

exports[`simple 16`] = `
-Infinity
`;

exports[`simple 2`] = `
null
`;

exports[`simple 3`] = `
undefined
`;

exports[`simple 4`] = `
[]
`;

exports[`simple 5`] = `
/xx/
`;

exports[`simple 6`] = `
Infinity
`;

exports[`simple 7`] = `
false
`;

exports[`simple 8`] = `
true
`;

exports[`simple 9`] = `
{}
`;

exports[`supports named snapshots: not so public knowledge 1`] = `
{
  "identity": "Batman",
  "name": "Bruce Wayne",
}
`;

exports[`supports named snapshots: public knowledge 1`] = `
{
  "name": "Bruce Wayne",
}
`;

exports[`supports named snapshots: public knowledge 2`] = `
{
  "address": "Arkham Asylum",
  "name": "Joker",
}
`;

exports[`test A 1`] = `
{
  "a": 20,
  "b": [
    1,
    2,
    "bar",
    5,
  ],
  "d": {
    "data": [
      102,
      111,
      111,
    ],
    "type": "Buffer",
  },
  "e": {
    "foo": "bar",
  },
}
`;

exports[`test B 1`] = `
{
  "__proto__": [],
  "b": [
    1,
    2,
    ,
    ,
    5,
  ],
  "e": {
    "foo": "bar",
  },
  "f": -Infinity,
}
`;

exports[`test B 2`] = `
[
  [Error: ?!],
  [TypeError: bar],
  Uint16Array [
    4,
    2,
    65123,
  ],
  null,
]
`;

exports[`test B 3`] = `
{
  "x": 1337,
}
`;

exports[`weird  names 
 1`] = `
"
"
`;

exports[`weird  names  !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_\`abcdefghijklmnopqrstuvwxyz{|}~ 1`] = `
" !"#$%&'()*+,-./0123456789:;<=>?@ABCDEFGHIJKLMNOPQRSTUVWXYZ[\\]^_\`abcdefghijklmnopqrstuvwxyz{|}~"
`;

exports[`weird  names $ 1`] = `
"$"
`;

exports[`weird  names > 1`] = `
">"
`;

exports[`weird  names \\
\`
\\\`\` 1`] = `
"\\
\`
\\\`\`"
`;

exports[`weird  names \\ 1`] = `
"\\"
`;

exports[`weird  names \` 1`] = `
"\`"
`;

exports[`weird  names \` \`
 \` 1`] = `
"\` \`
 \`"
`;

exports[`weird  names \` \` \` 1`] = `
"\` \` \`"
`;

exports[`weird  names multi
line 1`] = `
0
`;

exports[`weird  names with \` 1`] = `
42
`;

exports[`weird  names {} 1`] = `
"{}"
`;
